src/raps/unlockAndSwap.ts

```diff
@@ -101,24 +97,20 @@ export const createUnlockAndSwapRap = async (swapParameters: RapSwapActionParame
     from: accountAddress,
     sellTokenAddress,
     buyTokenAddress,
+    allowanceNeeded,
   } = quote as {
     from: Address;
     sellTokenAddress: Address;
     buyTokenAddress: Address;
+    allowanceNeeded: boolean;
   };
 
-  const isNativeAssetUnwrapping = isUnwrapNative({
-    chainId,
-    sellTokenAddress,
-    buyTokenAddress,
-  });
-
   // Aggregators represent native asset as 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE
   const nativeAsset = isLowerCaseMatch(ETH_ADDRESS_AGGREGATOR, sellTokenAddress) || isNativeAsset(sellTokenAddress, chainId);
 
   let swapAssetNeedsUnlocking = false;
 
-  if (!isNativeAssetUnwrapping && !nativeAsset) {
+  if (allowanceNeeded) {
```

Comment:
on line 123 you still have the `allowsPermit` logic in this file while it was removed in `unlockAndCrosschainSwaps` on line (121/104); wanted to confirm this

Reply:
This flow is still supported in singlechain swaps while it's not in crosschain swaps